nilearn/_utils/niimg_conversions.py

```diff
@@ -135,7 +162,10 @@ def concat_niimgs(niimgs, dtype=np.float32, accept_4d=False,
     """
 
     # get properties from first image
-    first_niimg = check_niimg(next(iter(niimgs)))
+    try:
+        first_niimg = check_niimg(next(iter(niimgs)), atleast_4d=True)
+    except StopIteration:
+        raise TypeError('Cannot concatenate empty objects')
```

Comment:
I kept the original behavior (this was checked higher in the stack before) but I agree with you. Maybe we can change that in another PR as I want this one to have the lowest impact possible on the logic of the code.